src/crypto/slow-hash.c

```diff
@@ -59,35 +61,136 @@ extern int aesb_pseudo_round(const uint8_t *in, uint8_t *out, const uint8_t *exp
   } while(0)
 
 #define VARIANT1_2(p) \
-  do if (variant > 0) \
+  do if (variant == 1) \
   { \
     xor64(p, tweak1_2); \
   } while(0)
 
 #define VARIANT1_CHECK() \
   do if (length < 43) \
   { \
-    fprintf(stderr, "Cryptonight variants need at least 43 bytes of data"); \
+    fprintf(stderr, "Cryptonight variant 1 needs at least 43 bytes of data"); \
     _exit(1); \
   } while(0)
 
 #define NONCE_POINTER (((const uint8_t*)data)+35)
 
 #define VARIANT1_PORTABLE_INIT() \
   uint8_t tweak1_2[8]; \
-  do if (variant > 0) \
+  do if (variant == 1) \
   { \
     VARIANT1_CHECK(); \
     memcpy(&tweak1_2, &state.hs.b[192], sizeof(tweak1_2)); \
     xor64(tweak1_2, NONCE_POINTER); \
   } while(0)
 
 #define VARIANT1_INIT64() \
-  if (variant > 0) \
+  if (variant == 1) \
   { \
     VARIANT1_CHECK(); \
   } \
-  const uint64_t tweak1_2 = variant > 0 ? (state.hs.w[24] ^ (*((const uint64_t*)NONCE_POINTER))) : 0
+  const uint64_t tweak1_2 = (variant == 1) ? (state.hs.w[24] ^ (*((const uint64_t*)NONCE_POINTER))) : 0
+
+#define VARIANT2_INIT() \
+  uint64_t division_result; \
+  uint32_t sqrt_result; \
+  do if (variant >= 2) \
+  { \
+    division_result = 0; \
+    sqrt_result = 0; \
+  } while (0)
```

Comment:
Probably faster (and simpler) to init them to 0 regardless.

Reply:
We have 200 bytes from initial keccak, it could be anything from those 200 bytes here if someone doesn't like zeroes.